codegen/src/main/java/software/amazon/awssdk/codegen/poet/waiters/BaseWaiterClassSpec.java

```diff
@@ -294,4 +328,143 @@ private OperationModel operationModel(WaiterDefinition waiterDefinition) {
     private String getWaiterMethodName(String waiterMethodName) {
         return "waitUntil" + waiterMethodName;
     }
+
+    private TypeName waiterAcceptorTypeName(WaiterDefinition waiterDefinition) {
+        WildcardTypeName wildcardTypeName = WildcardTypeName.supertypeOf(fullyQualifiedResponseType(waiterDefinition));
+
+        return ParameterizedTypeName.get(ClassName.get(List.class),
+                                         ParameterizedTypeName.get(ClassName.get(WaiterAcceptor.class), wildcardTypeName));
+    }
+
+    private TypeName fullyQualifiedResponseType(WaiterDefinition waiterDefinition) {
+        String modelPackage = model.getMetadata().getFullModelPackageName();
+        String operationResponseType = model.getOperation(waiterDefinition.getOperation()).getReturnType().getReturnType();
+        return ClassName.get(modelPackage, operationResponseType);
+    }
+
+    private CodeBlock acceptor(Acceptor acceptor) {
+        CodeBlock.Builder result = CodeBlock.builder();
+
+        switch (acceptor.getState()) {
+            case "success":
+                result.add("$T.success", WaiterAcceptor.class);
+                break;
+            case "failure":
+                result.add("$T.error", WaiterAcceptor.class);
+                break;
+            case "retry":
+                result.add("$T.retry", WaiterAcceptor.class);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported acceptor state: " + acceptor.getState());
+        }
+
+        switch (acceptor.getMatcher()) {
+            case "path":
+                result.add("OnResponseAcceptor(")
+                      .add(pathAcceptorBody(acceptor))
+                      .add(")");
+                break;
+            case "pathAll":
+                result.add("OnResponseAcceptor(")
+                      .add(pathAllAcceptorBody(acceptor))
+                      .add(")");
+                break;
+            case "pathAny":
+                result.add("OnResponseAcceptor(")
```

Comment:
Can this be move out of the switch? Looks like we always add except for `status` which doesn't use the `result` that's being built

Reply:
+1

Reply:
Actually, the error path is different, so I can't do this.